bobtimus/src/loan.rs

```diff
@@ -44,42 +45,520 @@ pub struct LoanOffer {
     /// The max_ltv protects the lender from Bitcoin falling too much.
     pub max_ltv: Decimal,
 
-    pub interest: Vec<Interest>,
+    /// Base interest in percent (to be applied to the principal amount)
+    pub base_interest_rate: Decimal,
+
+    /// Interest in relation to terms
+    pub terms: Vec<Term>,
+
+    /// Interest rates in relation to collteralization
+    pub collateralizations: Vec<Collateralization>,
 }
 
-#[derive(Debug, Clone, serde::Serialize)]
-pub struct Interest {
-    /// Loan term in days
-    pub term: u32,
-    /// Collateralization in percent
-    ///
-    /// Rational: If a borrower over-collateralizes with e.g. 150% -> better rate than at 140%
+#[derive(Debug, Clone, Copy, serde::Serialize)]
+pub struct Term {
+    pub days: u32,
+    /// Interest to be added on top of the base interest rate for this term
+    pub interest_mod: Decimal,
```

Comment:
What is this needed for?

Reply:
see https://github.com/comit-network/waves/pull/233#discussion_r687388697